flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -146,9 +146,9 @@ public static HiveObjectConversion getConversion(ObjectInspector inspector, Logi
 			} else if (inspector instanceof TimestampObjectInspector) {
 				conversion = hiveShim::toHiveTimestamp;
 			} else if (inspector instanceof HiveCharObjectInspector) {
-				conversion = o -> new HiveChar((String) o, ((CharType) dataType).getLength());
+				conversion = o -> o == null ? null : new HiveChar((String) o, ((CharType) dataType).getLength());
 			} else if (inspector instanceof HiveVarcharObjectInspector) {
-				conversion = o -> new HiveVarchar((String) o, ((VarCharType) dataType).getLength());
+				conversion = o -> o == null ? null : new HiveVarchar((String) o, ((VarCharType) dataType).getLength());
 			} else if (inspector instanceof HiveDecimalObjectInspector) {
 				conversion = o -> o == null ? null : HiveDecimal.create((BigDecimal) o);
 			} else {
```

Comment:
It's not dead code but WritableHiveObjectConversion is currently only used for constant arguments of UDFs.

Reply:
I see, this is because we change "getObjectInspectorForConstant" to writable hive type. Cna you add comments to explain?

Reply:
And we should deal with `null` in `WritableHiveObjectConversion` too.
Because this is a common util, not just for constant.

Reply:
> this is because we change "getObjectInspectorForConstant" to writable hive type

Do you mean the change in #10625? Actually even w/o #10625, this is not dead code because 1.1.x versions having been using writable OIs for constants.

> Cna you add comments to explain?

What do you think we should explain here? Please note that we may want to use writable OIs for both constant and non-constant in the future, as suggested in #10625.

> And we should deal with null in WritableHiveObjectConversion too

Let's leave that to #10625, since we're doing a bigger refactor there.